src/pages/workspace/distanceRates/PolicyDistanceRateEditPage.tsx

```diff
@@ -0,0 +1,201 @@
+                <FeatureEnabledAccessOrNotFoundWrapper
+                    policyID={policyID}
+                    featureName={CONST.POLICY.MORE_FEATURES.ARE_DISTANCE_RATES_ENABLED}
+                >
+                    <ScreenWrapper
+                        testID={PolicyDistanceRateEditPage.displayName}
+                        includeSafeAreaPaddingBottom={false}
+                        style={[styles.defaultModalContainer]}
+                        shouldShowOfflineIndicatorInWideScreen
+                    >
+                        <HeaderWithBackButton
+                            title={`${rateValueToDisplay} / ${translate(`common.${customUnit?.attributes?.unit ?? CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES}`)}`}
+                            shouldShowThreeDotsButton
+                            threeDotsMenuItems={threeDotsMenuItems}
+                            threeDotsAnchorPosition={styles.threeDotsPopoverOffset(windowWidth)}
+                        />
+                        <View style={styles.flexGrow1}>
+                            <OfflineWithFeedback
+                                errors={ErrorUtils.getLatestErrorField(rate, 'enabled')}
+                                pendingAction={rate?.pendingFields?.enabled}
+                                errorRowStyles={styles.mh5}
+                                onClose={() => clearErrorFields('enabled')}
+                            >
+                                <View style={[styles.flexRow, styles.justifyContentBetween, styles.p5]}>
+                                    <Text>{translate('workspace.distanceRates.enableRate')}</Text>
+                                    <Switch
+                                        isOn={rate.enabled ?? false}
+                                        onToggle={toggleRate}
+                                        accessibilityLabel={translate('workspace.distanceRates.enableRate')}
+                                    />
+                                </View>
+                            </OfflineWithFeedback>
+                            <OfflineWithFeedback
+                                errors={ErrorUtils.getLatestErrorField(rate, 'rate')}
+                                pendingAction={rate?.pendingFields?.rate}
+                                errorRowStyles={styles.mh5}
+                                onClose={() => clearErrorFields('rate')}
+                            >
+                                <MenuItemWithTopDescription
+                                    shouldShowRightIcon
+                                    title={`${rateValueToDisplay} / ${unitToDisplay}`}
+                                    description={translate('workspace.distanceRates.rate')}
+                                    descriptionTextStyle={styles.textNormal}
+                                    onPress={showRateModal}
+                                />
+                            </OfflineWithFeedback>
+                            <RateModal
```

Comment:
Im not sure I fully get whats the issue here. We have separate page for editing - `PolicyDistanceRateEditPage`. Once you click on the rate it opens up a modal to edit the rate - that modal is a separate component. It saves us creating new route and some navigating. You have an idea for different component structure?

Reply:
Wouldn't it be useful if the edit rate page/modal were accessible using a direct link? I don't feel strongly about this, but I wanted it to match other pages. your call 👍

Reply:
Looking at the categories I see what you mean. This view is accessible via direct link:
<img width="1920" alt="image" src="https://github.com/Expensify/App/assets/23487612/556a572a-585e-423e-b2cd-a771fd9e255e">

@luacmartins Do you think it's fine or should I add another route for edit form. In order to match categories with its design we would to add a new route and refactor current approach. It would be something like this:

```    
WORKSPACE_DISTANCE_RATE_DETAILS: {
        route: 'settings/workspace/:policyID/distance-rates/:rateID',
        getRoute: (policyID: string, rateID: string) => `settings/workspace/${policyID}/distance-rates/${rateID}/` as const,
    },
WORKSPACE_DISTANCE_RATE_EDIT: {
        route: 'settings/workspace/:policyID/distance-rates/:rateID/edit',
        getRoute: (policyID: string, rateID: string) => `settings/workspace/${policyID}/distance-rates/${rateID}/edit` as const,
    },
```

So the screen above would be accessible under `distance-rates/:rateID` route and the one below would be under
`distance-rates/:rateID/edit`

<img width="1920" alt="image" src="https://github.com/Expensify/App/assets/23487612/8710378e-9faf-4e30-b18b-6759a202922b">

Reply:
Yup 😅 No worries, I'm starting the refactor then